### ISSUE TITLE
Is Postage destined to be a community project?

### ISSUE THREAD

**Issue by justintocci** [workflowproducts/postage#32]
I had hoped that Postage would become a community project. 

Realistically, I doubt that will ever happen for many reasons. Some are pedestrian.
- The easier features are all complete. 
- The codebase is complex. 
- There are a lot of technologies that work together, such as C and javascript, that are not usually skills that are found in one programmer. In fact, many features were only implemented by having two people on our team work together.

Another is, we are a very small company with a very large (for us) investment in Envelope, the platform that Postage is built on. As such, we need to keep our options open. One way to do that would be to have outside contributors sign a waver basically giving up their rights to their contributions. I can imagine most people would find that unacceptable. How do we protect our future while giving contributors the proper expectations (whatever those might be)?

I think these realities make the barrier to entry pretty high. I'm not looking forward to figuring all of this out.

A friend of Postage mentioned that we could do a couple things to encourage patches from the community. One was to move the repo to postage/postage from workflowproducts/postage. Unfortunately, Postage as an organization is taken, but even if we chose an unoccupied organization name, many users bookmarks and clone sources would break if we changed organizations.

We actually did this once already and because it was poorly handled we lost all our stars and watches. 

Anyway, until we have some outside contributors I don't see a reason to put people through that. And then, if we do get some contributors, we'll likely have many more users. Why break all those links?

If the community has an opinion on this please let us know by posting here. We don't really know what to do so we'll probably just do what you want.

**Comment by anthonydb** [comment]
A couple of thoughts:

First, my previous experience on a team building a complex open source project (https://github.com/documentcloud) leads me to agree that as system complexity rises, the ability for people to jump in becomes harder. Especially when you're mixing in dependencies and development environment setup. We rarely got outside contributions to that project.

One way to encourage patches from the community, though, is to create issues tickets that a person could jump in on and label them "Easy." This could be items like documentation, UX cleanup etc.

**Comment by justintocci** [comment]
That is an excellent idea. Michael and Nunzio, please see if you've got any issues that qualify and label them. Let's all agree going forward to label easy fixes immediately and give the community a chance to get their feet wet.

**Comment by justintocci** [comment]
I am ready to close this issue.

After a brief hesitation at switching to an MIT license, I've discovered that the Apache license model continues to be better for us. 

To that end, I have added an Apache-style "Contributor License Agreement" to the project and added the appropriate references to it. We'll need to get a signed CLA from each contributor before accepting patches. I was originally kind of put off by this, but in researching it I realized it's a lot more common than I thought. 

I still don't think Postage will ever have many other contributors. But, since it seems to be gaining popularity, I'm not really worried about it anymore. 

Besides, Postage is really almost finished. We can keep adding bells and whistles but it's nearly at parity, feature-wise, with the parts of MS SQL Server Management Studio that we wanted. I think it will be a strong contribution to the PostgreSQL ecosystem for many years.
